lib/config/config-initializer.js

```diff
@@ -465,7 +465,7 @@ function promptUser(callback) {
                     default: "JavaScript",
                     choices: ["JavaScript", "YAML", "JSON"]
                 }
-            ], answers => {
+            ]).then(answers => {
                 try {
```

Comment:
Could we make this function return the Promise instead of using a callback with `try`/`catch`? I think mixing callbacks and Promises is an antipattern. (We could also probably switch to using a Promise [here](https://github.com/eslint/eslint/blob/8733e6a852ea9b49cef26650357598ca8c2c2a1f/bin/eslint.js#L64) so that we wouldn't need to deal with callbacks at all.)